tmt/convert.py

```diff
@@ -401,6 +410,17 @@ def read_nitrate(beaker_task, common_data, disabled):
             raise ConvertError(
                 "Unable to remove '{0}'.".format(md_path))
 
+    # Add environmental variables from Mafefile to those from Nitrate
+    if 'environment' in common_data.keys():
+        for case in individual_data:
+            if 'environment' in case.keys():
+                for env_var in common_data['environment'].keys():
+                    if env_var not in case['environment'].keys():
+                        case['environment'][env_var] = common_data['environment'][env_var]
+            else:
+                case['environment'] = common_data['environment']
+        common_data.pop('environment')
+
```

Comment:
Perhaps the logic above could be implemented using the `dict.update()` method?

    case_environment = case['environment']
    case['environment'] = common_data['environment'].copy()
    case['environment'].update(case_environment)

Seems to be a bit shorter and more readable?

Reply:
fixed in 3b9e5d8